### PR TITLE
User doc MP2 - first draft, following CC documentation

### PR DESCRIPTION
**Pull request by verena-neufeld**
MP2 user documentation, closely follows/shares text with coupled cluster user docs.
- I have added some references but I am sure more are missing. Please direct me to relevant papers if you think they should be included or feel free to delete references I have added.
- I'd also be happy to add more sections if desired.
- Maybe not part of this pull request, but in the future importing tested code snippets might be nice.

### CODE REVIEW

File: source/user/cc.rst [hunk 1]
```diff
@@ -6,6 +6,10 @@ Coupled-cluster theory
 
 *Modules*: :mod:`cc`, :mod:`pbc.cc`
 
+The interfaces to coupled cluster and MP2 calculations look very similar
+in PySCF.  Therefore, note the similarities of this section to the
```
Comment (susilehtola) [inline source/user/cc.rst:10]: "look very similar" -> "are similar"

File: source/user/cc.rst [hunk 2]
```diff
@@ -6,6 +6,10 @@ Coupled-cluster theory
...
+:ref:`theory_mp2` documentation.
```
Comment (susilehtola) [inline source/user/cc.rst:11]: You're repeating "similar" and "similarity". The latter could be replaced by "parallels"

File: source/user/mp.rst [hunk 3]
```diff
@@ -5,18 +5,134 @@ Second-Order Møller–Plesset Perturbation Theory
 
 *Modules*: :mod:`mp`, :mod:`pbc.mp`
 
+The interfaces to coupled cluster and MP2 calculations look very similar
+in PySCF.  Therefore, note the similarities of this section to the
+:ref:`theory_cc` documentation.
```
Comment (susilehtola) [inline source/user/mp.rst:10]: Same notes here

File: source/user/mp.rst [hunk 4]
```diff
@@ -5,18 +5,134 @@ Second-Order Møller–Plesset Perturbation Theory
...
+.. literalinclude:: ../../examples/mp/00-simple_mp2.py
+
+which outputs
+
+.. code::
+
+  converged SCF energy = -99.9873974403487
+  E(MP2) = -100.198764900659  E_corr = -0.211367460310054
+
+the Hartree--Fock energy, the MP2 energy and their difference, the
```
Comment (susilehtola) [inline source/user/mp.rst:31]: namely, the Hartree--Fock energy, ...

File: source/user/mp.rst [hunk 5]
```diff
@@ -5,18 +5,134 @@ Second-Order Møller–Plesset Perturbation Theory
...
+    pyscf.mp.MP2(mf).kernel()
+  
+  for the same result.
+
+
+Spin symmetry
+=============
+
+The MP2 module in PySCF supports a number of broken spin symmetry reference
+wavefunctions.  In particular, MP2 can be performed with a spin-restricted,
```
Comment (susilehtola) [inline source/user/mp.rst:49]: This would be clearer as

"reference wave functions with broken spin symmetry"

File: source/user/mp.rst [hunk 6]
```diff
@@ -5,18 +5,134 @@ Second-Order Møller–Plesset Perturbation Theory
...
+    from pyscf import grad
+    mygrad = mymp.Gradients()
+    grad = mygrad.kernel()
+
+
+Frozen orbitals
+===============
+
+By default, MP2 calculations are performed in PySCF without any frozen
+orbitals, including core orbitals. To freeze the lowest-energy core
```
Comment (susilehtola) [inline source/user/mp.rst:96]: By default, MP2 calculations in PySCF include all electrons; that is, also core orbitals are correlated.

File: source/user/mp.rst [hunk 7]
```diff
@@ -52,7 +168,8 @@ state, we can write a function to compute the correlation energy
 In this example, we concatenate :math:`\alpha` and :math:`\beta` orbitals to
 mimic the spin-orbitals.  After integral transformation, we zeroed out the
 integrals of different spin.  Here, the :mod:`ao2mo` module provides the general
-2-electron MO integral transformation.  Using this module, you are able to do
+2-electron MO integral transformation, using chemist's notation for the
```
Comment (susilehtola) [inline source/user/mp.rst:171]: chemists'

File: source/user/mp.rst [hunk 8]
```diff
@@ -52,7 +168,8 @@ state, we can write a function to compute the correlation energy
...
+integrals.  Using this module, you are able to do
```
Comment (susilehtola) [inline source/user/mp.rst:172]: are able to do -> can perform

File: source/user/ref_mp.bib [hunk 9]
```diff
@@ -0,0 +1,34 @@
+@article{Handy1985,
+  title   = {The elimination of singularities in derivative calculations},
+  journal = {Chemical Physics Letters},
```
Comment (susilehtola) [inline source/user/ref_mp.bib:3]: Chem. Phys. Lett.

File: source/user/ref_mp.bib [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+  volume  = {120},
+  number  = {2},
```
Comment (susilehtola) [inline source/user/ref_mp.bib:5]: Volume number not necessary

File: source/user/ref_mp.bib [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+  pages   = {151-158},
+  year    = {1985},
+  issn    = {0009-2614},
+  doi     = {https://doi.org/10.1016/0009-2614(85)87031-7},
```
Comment (susilehtola) [inline source/user/ref_mp.bib:9]: this is not a doi, remove https://doi.org/

File: source/user/ref_mp.bib [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+  author  = {Handy, N. C. and Amos, R. D. and Gaw, J. F. and Rice, J. E. and Simandiras, E. D.}
+}
+@article{Moller1934,
+  title     = {Note on an Approximation Treatment for Many-Electron Systems},
+  author    = {M\o{}ller, C. and Plesset, M. S.},
+  journal   = {Phys. Rev.},
+  volume    = {46},
+  issue     = {7},
+  pages     = {618--622},
+  numpages  = {0},
```
Comment (susilehtola) [inline source/user/ref_mp.bib:19]: delete

File: source/user/ref_mp.bib [hunk 13]
```diff
@@ -0,0 +1,34 @@
+@article{Handy1985,
+  title   = {The elimination of singularities in derivative calculations},
+  journal = {Chemical Physics Letters},
+  volume  = {120},
+  number  = {2},
+  pages   = {151-158},
```
Comment (susilehtola) [inline source/user/ref_mp.bib:6]: last page not necessary

File: source/user/ref_mp.bib [hunk 14]
```diff
@@ -0,0 +1,34 @@
...
+  doi     = {https://doi.org/10.1016/0009-2614(85)87031-7},
+  author  = {Handy, N. C. and Amos, R. D. and Gaw, J. F. and Rice, J. E. and Simandiras, E. D.}
+}
+@article{Moller1934,
+  title     = {Note on an Approximation Treatment for Many-Electron Systems},
+  author    = {M\o{}ller, C. and Plesset, M. S.},
+  journal   = {Phys. Rev.},
+  volume    = {46},
+  issue     = {7},
+  pages     = {618--622},
```
Comment (susilehtola) [inline source/user/ref_mp.bib:18]: issue and last page not necessary

File: source/user/ref_mp.bib [hunk 15]
```diff
@@ -0,0 +1,34 @@
...
+  publisher = {American Physical Society},
+  doi       = {10.1103/PhysRev.46.618}
+}
+@article{Pople1979,
+  author  = {Pople, J. A. and Krishnan, R. and Schlegel, H. B. and Binkley, J. S.},
+  title   = {Derivative studies in hartree-fock and møller-plesset theories},
+  journal = {International Journal of Quantum Chemistry},
+  volume  = {16},
+  number  = {S13},
+  pages   = {225-241},
```
Comment (susilehtola) [inline source/user/ref_mp.bib:31]: number and last page not necessary

File: source/user/ref_mp.bib [hunk 16]
```diff
@@ -0,0 +1,34 @@
...
+  doi     = {https://doi.org/10.1002/qua.560160825},
```
Comment (susilehtola) [inline source/user/ref_mp.bib:32]: remove spurious https://doi.org/

File: source/user/ref_mp.bib [hunk 17]
```diff
@@ -0,0 +1,34 @@
...
+  pages     = {618--622},
+  numpages  = {0},
+  year      = {1934},
+  month     = {Oct},
+  publisher = {American Physical Society},
+  doi       = {10.1103/PhysRev.46.618}
+}
+@article{Pople1979,
+  author  = {Pople, J. A. and Krishnan, R. and Schlegel, H. B. and Binkley, J. S.},
+  title   = {Derivative studies in hartree-fock and møller-plesset theories},
```
Comment (susilehtola) [inline source/user/ref_mp.bib:27]: Capitalization

**Comment by gkc1000** [comment]
Susi - since I merged, in this case I think you can directly edit

File: source/user/ref_mp.bib [hunk 18]
```diff
@@ -0,0 +1,34 @@
...
+  journal = {International Journal of Quantum Chemistry},
```
Comment (susilehtola) [inline source/user/ref_mp.bib:28]: Int. J. Quantum Chem.

**Pull request merged**